nasa-gibs/worldview
# WV-2644-mobile-tooltips

## Description

Fixes: When you load an event that has event tracks, like icebergs or tropical storms, if you hover over a point, you get a tooltip showing either the date and area of the iceberg or date and speed values. If you try to view these on mobile, since there is no hover state, it's difficult to retrieve those tooltips. You can tap on the point and the tooltip will appear, but then the imagery will update and the tooltip will disappear.

In addition, you should be able to hover over the event icon on the map, and it should show you the name of the event. It's not possible to get this to show at all.

We should think of an elegant solution where we'd be able to bring up those tooltips easily and not have them disappear so quickly. 

## How To Test

1. `git checkout WV-2679-layer-picker-menu-header-bug`
2. `npm ci`
3. `npm run watch`
4. Got to `http://localhost:3000`
5. Use the developer tools to mock a mobile device without any device capable of a hover interaction.
6. Click on the "Events" tab.
7. Select an event with a path like a hurricane or an iceberg.
8. Click on a waypoint.
9. The tooltip should display and stay open.

## PR Submission Checklist

This is simply a reminder of what we are going to look for before merging your code.

- I have read the [CONTRIBUTING](https://github.com/nasa-gibs/worldview/blob/main/.github/CONTRIBUTING.md) doc
- I have added the necessary documentation (if applicable)
- I have added tests that prove my fix is effective or that my feature works (if applicable)
- Any dependent changes have been merged and published in downstream modules (if applicable)

## Merging

Please use the `squash and merge` commit method unless each commit in your branch is vital to the commit history of main.

@nasa-gibs/worldview
